sr_robot_commander/test/test_robot_commander.py

```diff
@@ -0,0 +1,652 @@
+            robot_state = RobotState()
+            for key, value in CONST_RA_HOME_ANGLES .items():
+                robot_state.joint_state.name.append(key)
+                robot_state.joint_state.position.append(value)
+            plan = self.robot_commander.plan_to_named_target(target_names[0], robot_state)
+
+        self.assertTrue(type(self.robot_commander._SrRobotCommander__plan) == RobotTrajectory)
+
+    def test_plan_to_named_target_target_not_exists(self):
+        self.reset_to_home()
+        const_test_name = "test_non_existing_target"
+        self.robot_commander.plan_to_named_target(const_test_name, None)
+        condition = self.robot_commander._SrRobotCommander__plan is None
+        self.assertTrue(condition)
+
+    def test_get_named_targets(self):
+        self.assertTrue(type(self.robot_commander.get_named_targets()) == list)
+
+    def test_get_joints_position(self):
+        ret_val = self.robot_commander.get_joints_position()
+        self.assertTrue(type(ret_val) == dict)
+
+    def test_get_joints_velocity(self):
+        ret_val = self.robot_commander.get_joints_velocity()
+        self.assertTrue(type(ret_val) == dict)
+
+    def test_get_joints_state(self):
+        ret_val = self.robot_commander.get_joints_state()
+        self.assertTrue(type(ret_val) == JointState)
+
+    def test_run_joint_trajectory(self):
+        self.reset_to_home()
+        trajectory = self.robot_commander.plan_to_joint_value_target(CONST_RA_HOME_ANGLES, angle_degrees=False,
+                                                                     custom_start_state=None).joint_trajectory
+        self.robot_commander.run_joint_trajectory(trajectory)
+        end_state = copy.deepcopy(self.robot_commander.get_current_state())
+        condition = self.compare_joint_states(CONST_RA_HOME_ANGLES, end_state)
+        self.assertTrue(condition)
+
+    def test_make_named_trajectory(self):
+        self.reset_to_home()
+        trajectory = []
+        trajectory.append({"joint_angles": {'ra_shoulder_pan_joint': 0.00, 'ra_elbow_joint': 2.00},
+                           "interpolate_time": 0.5, "pause_time": 0.1, "degrees": False})
+        trajectory.append({"joint_angles": {'ra_shoulder_pan_joint': 0.05, 'ra_elbow_joint': 2.00},
+                           "interpolate_time": 0.4, "pause_time": 0.1, "degrees": False})
+        trajectory.append({"joint_angles": {'ra_shoulder_pan_joint': 0.05, 'ra_elbow_joint': 1.95},
+                           "interpolate_time": 0.6, "pause_time": 0.1, "degrees": False})
+        trajectory.append({"joint_angles": {'ra_shoulder_pan_joint': 0.10, 'ra_elbow_joint': 1.90},
+                           "interpolate_time": 0.7, "pause_time": 0.1, "degrees": False})
+        traj = self.robot_commander.make_named_trajectory(trajectory)
+        all_positions = []
+
+        for i in range(0, len(traj.points)):
+            all_positions.append(traj.points[i].positions)
+        for wp in trajectory:
```

Comment:
whats wp?

Reply:
waypoint, the same 'naming convention' for it as from sr_robot_commander https://github.com/shadow-robot/sr_interface/blob/noetic-devel/sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py#L575